packages/react-flex/Spec.md

```diff
@@ -0,0 +1,524 @@
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <span grow>wide</span>
+</Flex>
+```
+
+### Shrinking
+
+#### Globally
+
+```HTML
+<Flex shrink={false}>
+  <span>wide</span>
+  <span>wide</span>
+</Flex>
+```
+
+#### Individually
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <span shrink={false}>wide</span>
+</Flex>
+```
+
+### Complex example
+
+```HTML
+<Flex
+  as="section"
+  direction="row-reverse"
+  horizontalAlign="space-between"
+  verticalAlign="center"
+  gap="5rem"
+  wrap grow inline>
+  <span order={3}>3</span>
+  <span shrink={false}>1</span>
+  <span basis="25%">2</span>
+</Flex>
+```
+
+## Migration
+
+_Describe what will need to be done to upgrade from the existing implementations:_
+
+### _Migration from v8_
+
+#### Stack
+
+##### `disableShrink`
+
+Renamed from `disableShrink` to `shrink`
+
+Before
+
+```HTML
+<Stack disableShrink>
+ items
+</Stack>
+```
+
+After
+
+```HTML
+<Flex shrink="0">
+  items
+</Flex>
+```
+
+```HTML
+<Flex shrink={false}>
+  items
+</Flex>
+```
+
+##### `horizontal`
+
+Before
+
+```HTML
+<Stack horizontal>
+ items
+</Stack>
+```
+
+After
+
+```HTML
+<Flex>
+  items
+</Flex>
+```
+
+##### `childrenGap`
+
+Renamed from `childrenGap` to `gap`
+
+##### `reversed` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack reversed>
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex direction="column-reverse">
+  items
+</Flex>
+```
+
+##### `maxHeight` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack maxHeight="30px">
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex style={{maxHeight:"30px"}}>
+  items
+</Flex>
+```
+
+##### `maxWidth` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack maxWidth="30px">
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex style={{maxWidth:"30px"}}>
+  items
+</Flex>
+```
+
+##### `verticalFill` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack verticalFill>
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex style={{height: "100%"}}>
+  items
+</Flex>
+```
+
+##### `padding` [DEPRECATED]
+
+#### StackItem
+
+##### `disableShrink`
+
+Renamed from `disableShrink` to `shrink`
+
+Before
+
+```HTML
+<Stack>
+ <Stack.Item grow>item</StackItem>
+ <Stack.Item grow disableShrink>item</StackItem>
+</Stack>
+```
+
+After
+
+```HTML
+<Flex>
+  <Flex.Item grow>item</Flex.Item>
+  <Flex.Item grow shrink="0">item</Flex.Item>
+</Flex>
+```
+
+```HTML
+<Flex>
+  <Flex.Item grow>item</Flex.Item>
+  <Flex.Item grow shrink={false}>item</Flex.Item>
+</Flex>
+```
+
+##### `verticalFill` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack>
+ <Stack.Item verticalFill>item</Stack.Item>
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex>
+ <Flex.Item style={{height: "100%"}}>item</Flex.Item>
+</Flex>
+```
+
+### _Migration from v0_
+
+#### Flex
+
+##### `column`
+
+Before
+
+```HTML
+<Flex column>
+ items
+</Flex>
+```
+
+After
+
+```HTML
+<Flex direction="column">
+  items
+</Flex>
+```
+
+##### `hAlign`
+
+Renamed from `hAlign` to `horizontalAlign`
+
+##### `vAlign`
+
+Renamed from `vAlign` to `verticalAlign`
+
+##### `debug` [DEPRECATED]
```

Comment:
**q:** does it mean that new implementation will still support this API ?

Reply:
For this case, it means were keeping the same usage (same input values), but the prop is now called `verticalAlign` for coherence with `Stack` and because I think the extra verbosity makes it much more clearer to our users than simply a `v` or an `h`.
Should I add extra text here too reinforcing that implementation remains the same? This is already described in the API Proposal section, but perhaps not enough in depth that it might come across as confusing. What do you think?

Reply:
oh, my question was about `##### debug [DEPRECATED]` :D

Reply:
ah nevermind, completely forgot that this chapter is solely to `Migration`, so the answer is NO, `debug` will not be a part of new API. thx!